### PR TITLE
Added docs sections and started with components.

### PR DESCRIPTION
**Pull request by blackbaud-brandonhare**

### CODE REVIEW

File: skyuxconfig.json [hunk 1]
```diff
@@ -1,4 +1,11 @@
 {
+  "name": "stache2",
   "mode": "easy",
-  "compileMode": "aot"
+  "compileMode": "aot",
+  "plugins": [
+    "@blackbaud/skyux-builder-plugin-stache-code-block"
+  ],
+  "app": {
+    "title": "Stache 2"
```
Comment (Blackbaud-BrandonJones) [inline skyuxconfig.json]: Should this be Stache 2 Docs?

File: src/app/app-extras.module.ts [hunk 2]
```diff
@@ -10,7 +12,9 @@ import { StacheModule } from '@blackbaud/stache';
   exports: [
     StacheModule
   ],
-  providers: [],
+  providers: [
+    StacheDemoComponentService
```
Comment (Blackbaud-BrandonJones) [inline src/app/app-extras.module.ts]: Should we move this into the Stache Module and just leave it undocumented?   Less confusion for people looking at our docs as an example.

File: src/app/components/action-buttons/action-buttons-demo.component.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      summary: 'Basic page layouts using Stache and skyux pattern, for quick page layouts.'
+    },
+    {
+      name: 'Layout: Sidebar',
+      path: '/components/layout-sidebar',
+      icon: 'columns',
+      summary: 'Side bar navigation layout pattern for creating quick documentation.'
+    }
+  ];
+}
```
Comment (Blackbaud-BrandonJones) [inline src/app/components/action-buttons/action-buttons-demo.component.ts]: I still really don't like the idea of having any components at all inside the docs.   This seems like stuff we really should be abstracting away.

File: src/app/components/demo-component.service.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+      if (nameA > nameB) {
+          return 1;
+      }
+      return 0;
+    });
+  }
+
+  public getByName(name: string): StacheDemoComponent {
+    return this.components.filter(component => component.name === name)[0];
+  }
```
Comment (Blackbaud-BrandonJones) [inline src/app/components/demo-component.service.ts]: This file should majority be gone I think....

Comment (blackbaud-brandonhare) [inline src/app/components/demo-component.service.ts]: We still need it for the action buttons on the components page

File: skyuxconfig.json [hunk 5]
```diff
@@ -1,4 +1,11 @@
 {
+  "name": "stache2",
```
Comment (Blackbaud-BrandonJones) [inline skyuxconfig.json]: not sure what this name is used for, but should it also be stache2Documentation?

File: src/app/components/action-buttons/action-buttons-demo.component.html [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    <stache-action-buttons [routes]="routes">
+    </stache-action-buttons>
+  </div>
+
+  <stache-page-anchor>
+    Code
+  </stache-page-anchor>
+  <div class="stache-demo-code">
+    <sky-tabset>
+      <sky-tab [tabHeading]="'action-buttons-demo-component.html'">
```
Comment (Blackbaud-SteveBrush) [inline src/app/components/action-buttons/action-buttons-demo.component.html]: `<sky-tab tabHeading="action-buttons-demo-component.html">`

**Pull request merged**